### PR TITLE
Make things rounder and stop screaming at users

### PR DESCRIPTION
**Pull request by oakkitten**
For the most part this makes various popup corners rounder and removes button labels in ALL CAPS. Roundness now follows the default Android 11 looks. Also, menus with icons were tweaked to have better looking padding and regular colors. I also changed a few things around the stuff I touched. As usual, please see individual commit messages for details.

This does not cover everything, particularly in places that might require significant overhaul. Notably,

* Filter by flag in Card browser menu—hopefully this is turned into a chip;
* Editor Tags & Cards buttons—these probably better be reworked into something else;
* Advanced editor—I think it should be removed altogether;
* Crop image—it does not follow app theming at all;
* Deck options—this one is fine on recent Android versions but shows jagged edges on 6. It would probably be better to deal with this by rooting out all the hacky stuff from it;
* TEXT in Statistics → Overview;
* Probably I missed something else?

Anyway, screenshots. Left: before; right: after.

I suppose the next big step would be to fully move to material theming. At this point gradual changes become counterproductive; I'd love to rip out every customization, switch to material and rebuild the UI top down. Big big work. Some of the above points should be addressed first though.

Accessibility scanner complains about the white text on blue/gray background in tab layout. This is an ongoing problem with the toolbar color being too light in light themes and should be solved by changing toolbar color.

## How Has This Been Tested?

Manually tested on my Xiaomi Mi A2 (Android 11) and Motorola Moto G 1st Gen (Android 6).

## Checklist

- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by BrayanDSO** [comment]
Overall, I really like the changes and agree with the path to take about the UI. I don't have time to review it right now, but I should do it on the weekend if nobody else beats me up.

About the points on the PR description, I agree with them, although I don't recommend spending time on Deck options or Statistics, as they eventually are going to be replaced by upstream's HTML pages when the new backend is the default

And I would include Filtered Deck options on the pending list, which is quite hacky as well

Also this have an Lint issue

**Comment by oakkitten** [comment]
Updated to fix the lint issue. Also I found a menu with flags in Card browser so I dealt with that as well, and I decided that at this point it's better to go recursive with fixing up the menus.

File: AnkiDroid/src/main/java/com/ichi2/utils/MenuUtils.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        item.icon?.let { icon ->
+            if (icon !is Wrapper) item.icon = Wrapper(icon)
+        }
+    }
+}
+
+/**
+ * Recursively tints the icons of the items of the given overflow or popup menu
+ * with the color specified in the theme. Has no effect for items that have no icon.
+ */
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/utils/MenuUtils.kt:65]:
```suggestion
/**
 * Recursively tints the icons of the items of the given overflow or popup menu
 * with the color specified in the theme with [R.attr.overflowAndPopupMenuIconColor]. 
 * Has no effect for items that have no icon.
 */
```

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt [hunk 2]
```diff
@@ -230,103 +230,95 @@ class ModelBrowser : AnkiActivity() {
         mActionBar!!.subtitle = resources.getQuantityString(R.plurals.model_browser_types_available, count, count)
     }
 
-    /*
-     *Creates the dialogue box to select a note type, add a name, and then clone it
+    /**
+     * This:
+     *
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt:235]: unnecessary. The javadoc is for the method, so clearly it's speaking about it

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt [hunk 3]
```diff
@@ -230,103 +230,95 @@ class ModelBrowser : AnkiActivity() {
...
-     */
-    @KotlinCleanup("Use scope function while initializing oldModel + Invert and return early")
-    private fun addNewNoteType(modelName: String, position: Int) {
-        val model: Model
-        if (modelName.isNotEmpty()) {
-            val nbStdModels = StdModels.STD_MODELS.size
-            model = if (position < nbStdModels) {
-                StdModels.STD_MODELS[position].add(mCol)
+        // TODO This will happily accept names that already exist in the user model list.
+        //   If duplicate names are not undesirable, add a comment stating so.
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt:285]: just flagging so someone remembers to create an issue later

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt [hunk 4]
```diff
@@ -230,103 +230,95 @@ class ModelBrowser : AnkiActivity() {
...
-        }
-        for (model in mModels!!) {
-            val name = model.getString("name")
-            mNewModelLabels!!.add(String.format(clone, name))
-            mNewModelNames!!.add(name)
-            mExistingModelNames!!.add(name)
+        // TODO These dialogs are slightly confusing.
+        //   The first one says “Add note type ... Ok”, but it doesn't add anything straight away.
+        //   The second one also says “Add note type ... Ok”, and it is not mention the model
+        //   that you are cloning. I suggest reworking the strings so this is less confusing.
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt:309]: ditto

**Review by BrayanDSO — CHANGES_REQUESTED** [review]
Overall, LGTM. My review comments are only nits.

The only thing that is blocking to me is the bottom navigation's pill background contrast (I guess that the same may apply to the night themes).

image
image

I know that before the PR, it didn't pass the accessibility scanner either, but this makes the contrast ratio even lower and it's best to move forward.

**Comment by snowtimeglass** [comment]
The rounder corners look good.

>Also, menus with icons were tweaked to have better looking padding and regular colors. 

@oakkitten Personally, about the color of the icons of the menu of Study screen, I like and am attached to the current blue one, 
image
but, as you probably think, it may not be the standard style of the latest Material Design. I've thought so since seeing the following the designs of Google Chrome and Gmail. I've begun to prepare myself to say goodbye to the current blue icons.
image

>Accessibility scanner complains about the white text on blue/gray background in tab layout. This is an ongoing problem with the toolbar color being too light in light themes and should be solved by changing toolbar color.

I support the solution like the following one, using white or pale gray or pale blue. Google Chrome and Gmail do so. 

>- White theme with dark text (instead of every theme having light text)

_Originally posted by @nickdvlpr in https://github.com/ankidroid/Anki-Android/issues/5126#issuecomment-451238994_

image



image

_Originally posted by @nickdvlpr in https://github.com/ankidroid/Anki-Android/issues/5126#issue-383325736_

Blue toolbar doesn't seem to suit the dark gray icons (even if the color of the bar becomes darker). 
image

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt (hunk 2 above)
Comment (oakkitten) [reply AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt]: Done

File: AnkiDroid/src/main/java/com/ichi2/utils/MenuUtils.kt (hunk 1 above)
Comment (oakkitten) [reply AnkiDroid/src/main/java/com/ichi2/utils/MenuUtils.kt]: Done, I slightly changed the wording though

**Comment by oakkitten** [comment]
Rebased to address some of the suggestions. I have absolutely no idea what's going on with the failing unit tests here.

> The only thing that is blocking to me is the bottom navigation's pill background contrast

This is a problem, but I think this change is a step in the right direction. Previously it wasn't immediately clear which item is selected, as you can have both darker and lighter color for selection. Of course, the big clue is that of the three items only one can be selected, but this is still confusing. This changes the behavior to a more standard, Android way, and the problem will disappear once the toolbar color is changed to a light color like @snowtimeglass mentions.

> but, as you probably think, it may not be the standard style of the latest Material Design. I've thought so since seeing the following the designs of Google Chrome and Gmail. I've begun to prepare myself to say goodbye to the current blue icons.

I think I would prefer the icons to be the same color as text. Blue toolbar indeed seems to be not agreeing with the dark grey icons somewhat, but if the icons are the same color as text, there's the agreement! I definitely don't like the blue color though. I didn't change it all the way to the color I like because this is probably best done as a part of a PR that changes the entire theme. I didn't want to make such decisions just yet so opted for what seems to be the more standard and uncontroversial color.

**Comment by BrayanDSO** [comment]
1: CI is broken

2: I'm okay with the light pill color depending on the time frame of changing the bars colors, as I'm already foreseeing someone complaining about the low contrast in the future. This will probably take some time/discussion, as this is a "big" change on the app appearance. I think that there is a `colorPrimaryDark` attribute already chosen for `Light` and `Plain` themes, which may be a good temporary solution while the bar's color isn't changed.

**Comment by oakkitten** [comment]
I was thinking about inverting the colors, displaying a fully white pill and making the icon itself blue. I didn't go with it as this effectively makes the icon appear flipped. These icons need to be changed anyway, I guess I just might draw some that can be inverted, how about this? I planned to do this later but might as well do it now.

**Comment by BrayanDSO** [comment]
👍🏻

**Comment by oakkitten** [comment]
Updated to include this. What do you think?

**Comment by BrayanDSO** [comment]
Fine by me. I've reran CI tests. 

Optional: have you tried using filled icons when they are selected? (as Material recommends https://m3.material.io/components/navigation-bar/overview)

**Comment by oakkitten** [comment]
I don't know what's happening with CI, tests pass (on ubuntu/macos) [in my repo](https://github.com/oakkitten/Anki-Android/actions/runs/3122322628).

I thought about filled icons, but
  * It would be better for consistency to make the CSS icon behave similarly. I am not sure what'd be a good way to do that, simply outlining the letters CSS seems like a bad idea;
  * TBH I just don't want to deal with this right now, and it can be done later anyway.

**Comment by BrayanDSO** [comment]
Odd. I've rerun 5 times I think? My PC is currently busy, so I can't run it locally for the moment.

About the filled icons, I don't think it's bad to have one icon that is not outlined (in this case, CSS). You can have Google Maps as an example. But that's an issue for another day.

**Comment by BrayanDSO** [comment]
@oakkitten finally got to run this on my PC and the same errors appear here.

**Comment by oakkitten** [comment]
Rebased onto current main, also changing `DrawableWrapper` to `DrawableWrapperCompat` to appease CI.

**Review by BrayanDSO — APPROVED** [review]

**Comment by BrayanDSO** [comment]
CI fails

**Comment by mikehardy** [comment]
I think the before / afters are a tangible user-visible improvement and will go a long way towards reducing the "Ugh, this app looks like it's from 2008" complaints on our play store reviews

Needs a compile fix 

>  e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt: (129, 17): Unresolved reference: addNewNoteTypeDialog

**Comment by mikehardy** [comment]
The conflict has somehow survived your force push :thinking:

**Comment by oakkitten** [comment]
It was another one. 🤯 Hope it's good now

**Review by mikehardy — APPROVED** [review]
Sam Elliot approves

hat-tip-gif-6

**Pull request merged**